qiskit/_jobprocessor.py

```diff
@@ -0,0 +1,366 @@
+                        formatted_circuits.append(openquantumcompiler.dag2json(circuit))
+                else:
+                    for circuit in self.circuits:
+                        formatted_circuits.append(circuit.qasm(qeflag=True))
+            # create circuit component of qobj
+            circuitRecords = []
+            if circuit_configs is None:
+                config = {'coupling_map': None,
+                          'basis_gates': 'u1,u2,u3,cx,id',
+                          'layout': None,
+                          'seed': seed}
+                circuit_configs = [config] * len(self.circuits)
+            for circuit, fcircuit, name, config in zip(self.circuits,
+                                                       formatted_circuits,
+                                                       self.names,
+                                                       circuit_configs):
+                record = {
+                    'name': name,
+                    'compiled_circuit': None if doCompile else fcircuit,
+                    'compiled_circuit_qasm': None if doCompile else fcircuit,
+                    'circuit': circuit,
+                    'config': config
+                }
+                circuitRecords.append(record)
+            qobjid = ''.join([random.choice(
+                string.ascii_letters + string.digits) for i in range(10)])
+            self.qobj = {'id': qobjid,
+                         'config': {
+                             'max_credits': resources['max_credits'],
+                             'shots': shots,
+                             'backend': backend
+                         },
+                         'circuits': circuitRecords
+            }
+        self.seed = seed
+        self.result = None
+        self.doCompile = doCompile
+
+
+class JobProcessor():
+    """
+    process a bunch of jobs and collect the results
+    """
+    def __init__(self, q_jobs, callback, max_workers=1, token=None, url=None, api=None):
+        """
+        Args:
+            q_jobs (list(QuantumJob)): List of QuantumJob objects.
+            callback (fn(results)): The function that will be called when all
+                jobs finish. The signature of the function must be:
+                fn(results)
+                results: A list of (result, qobj) tuples.
+            max_workers (int): The maximum number of workers to use.
+            token (str): Server API token
+            url (str): Server URL.
+            api (IBMQuantumExperience): API instance to use. If set,
+                /token/ and /url/ are ignored.
+        """
+        self.q_jobs = q_jobs
+        self.max_workers = max_workers
+        # check whether any jobs are remote
+        self.online = False
+        self._local_backends = local_backends()
+        for qj in self.q_jobs:
+            if qj.backend not in self._local_backends:
+                self.online = True
+                break
```

Comment:
If you want to make this loop an one-liner, it could be written in a similar way as:
```python
self.online = any(qj.backend not in self._local_backends for qj in q_jobs)
```
Not sure if it is more readable that way, though!

Reply:
great! love it...